grant-project/zcash-grant-system
# Settings redesign + change email + change password conf email

Closes #48, and makes some improvements along the way. Please merge #107 in BEFORE merging this, or merge this unsquashed.

### What This Does

* Adds the ability to change your email. Resets your email confirmation, and sends an email to confirm at your new address, as well as a notification to your old address.
* Adds an email send to your email when you change your password. This is just standard security best practice.
* Redesigns the settings page to be more space efficient as we add more tabs, and gives it a tighter mobile layout. Let me know what you think @AMStrix 
* Renames some profile endpoints to use the `*/me/*` convention, since they weren't particularly RESTful PUT endpoints given a lack of resource id.

### What this Doesn't Do

* Reset auth tokens. @AMStrix I didn't know too much about the flask security conventions here, let me know if / how this should be done.

### Steps to Test

* Go to your /settings, and change your email. Confirm it validated input correctly (bad email, incorrect password, same email etc.)
    * You can just add `*+etc@gmail.com` if you have a gmail.
* Confirm your old email got an email notification of the change.
* Confirm your new email got an email to confirm it.
* Confirm the email confirmation link works.
* Change your password, confirm your new email got the pw change confirmation.

### Screenshots

<img width="600" alt="screen shot 2019-01-24 at 2 08 29 pm" src="https://user-images.githubusercontent.com/649992/51703399-3ee96b00-1fe4-11e9-9f8e-0a7f94f44b5d.png">


<img width="280" alt="screen shot 2019-01-24 at 2 19 02 pm" src="https://user-images.githubusercontent.com/649992/51703410-427cf200-1fe4-11e9-9d4a-f75e1a9cf550.png">
